observer-errors/README.md

```diff
@@ -0,0 +1,130 @@
+
+### Expected Exceptions in the Observer API
+
+There are at least three kinds of errors to be considered in the Observer model:
+
+#### Configuration
+
+Those are the exceptions thrown when you're setting up a new instance or setting elements to be observed, e.g.: on the immediate call for `o.observe(elem)`. Those are immediate errors and can be be captured today.
+
+Another example would be using an invalid argument for the `ResizeObserver`'s callback function. Users can handle this exception today, there is nothing to be done here.
+
+```js
+new ResizeObserver({}); // TypeError
+```
+
+#### <a id="effects"></a> Effects within the observable elements
+
+Errors due to the resize effect of observable elements and the resize loop (over the boundaries, etc).
+
+Some of these errors may already be recognized by each of Observer API's explainer, like the ResizeObserver's [Deliver Resize Loop error notification](https://drafts.csswg.org/resize-observer-1/#deliver-resize-error).
+
+#### Callback executions
+
+Execution errors when the Observer callback function is called. When unhandled, they bubble out of the callback function execution and eventually hit as an window error.
+
+One way to exemplify this would be an immediate throwing expression:
+
+```js
+new ResizeObserve(function() {
+  throw new Error();
+});
+```
+
+The error from the callback function cannot be captured from registering the elements through the `.observe` method. The registration does not imply a call to that function.
+
+```js
+const observer = new ResizeObserver(function() {
+    throw new Error();
+});
+
+try {
+    observer.observe(elem1);
+} catch (e) {
+    /* the only error to be captured here is if elem1 don't meet the parameters requirements */
+}
+```
+
+The current Observer model has no control over the given error. This error would bubble out and eventually captured in the `window.onerror`. It would be useful for the Observer instances to provide some control to capture and/or handle the error.
+
+While the `.onerror` remains the classic approach for handling errors in window and elements, it doesn't fit well in the Observer the same way it can't add an event listener as in `window.addEventListener`. This matches the idea on [observers API not being based in events](https://github.com/WICG/resize-observer/blob/master/explainer.md#why-an-observer-based-api-and-not-events).
+
+This proposal tries a less complex solution that uses another callback for catching the errors.
+​
+```js
+var observer = new ResizeObserver(function(entries) { /* ... */ });
+observer.observe(elem1);
+observer.observe(elem2);
+observer.observe(elem3);
+
+// The new proposed method
+observer.catch(function(err) {
+    /* error to be captured when the callback is executed */
```

Comment:
If the error handler is on the overall observer, should it indicate which element(s) had the error?

Reply:
I believe my answer above (https://github.com/salesforce/standards-explainers/pull/2/files#r600184788) also applies here.

So far, the catch just reflects the error it captures. This way, errors should be defined elsewhere, specific to each observer, and providing information on the element. See the "Effects within the observable elements" part of this doc. It also links to one possible [error for the Resize Observer loop](https://drafts.csswg.org/resize-observer-1/#deliver-resize-error). In this specific case, I don't think an element is reported.